packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart.ts

```diff
@@ -1,242 +0,0 @@
-/*
- * Copyright (c) 2020, salesforce.com, inc.
- * All rights reserved.
- * Licensed under the BSD 3-Clause license.
- * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
- */
-
-import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
-import * as path from 'path';
-import * as vscode from 'vscode';
-import { channelService } from '../../channels';
-import { nls } from '../../messages';
-import { notificationService, ProgressNotification } from '../../notifications';
-
-import { taskViewService } from '../../statuses';
-import { telemetryService } from '../../telemetry';
-import { OrgAuthInfo } from '../../util';
-import {
-  FilePathGatherer,
-  SfdxCommandlet,
-  SfdxWorkspaceChecker
-} from '../util';
-
-import { Uri, window } from 'vscode';
-import { FunctionService } from './functionService';
-import {
-  FUNCTION_DEFAULT_DEBUG_PORT,
-  FUNCTION_DEFAULT_PORT,
-  FUNCTION_RUNTIME_DETECTION_PATTERN
-} from './types/constants';
-
-import { getFunctionsBinary } from '@heroku/functions-core';
-import { getProjectDescriptor } from '@heroku/functions-core';
-import { LibraryCommandletExecutor } from '@salesforce/salesforcedx-utils-vscode/out/src';
-import { OUTPUT_CHANNEL } from '../../channels';
-
-const LOG_NAME = 'force_function_start';
-
-/**
- * Error types when running SFDX: Start Function
- * This is also used as the telemetry log name.
- */
-type ForceFunctionStartErrorType = 'force_function_start_docker_plugin_not_installed_or_started';
-
-const forceFunctionStartErrorInfo: {
-  [key in ForceFunctionStartErrorType]: {
-    cliMessage: string;
-    errorNotificationMessage: string;
-  };
-} = {
-  force_function_start_docker_plugin_not_installed_or_started: {
-    cliMessage: 'Cannot connect to the Docker daemon',
-    errorNotificationMessage: nls.localize(
-      'force_function_start_warning_docker_not_installed_or_not_started'
-    )
-  }
-};
-
-export class ForceFunctionStartExecutor extends LibraryCommandletExecutor<
```

Comment:
Refactored this into a base abstract class that is extended by the container and containerless executors.  There was a large chunk of shared code which drove my decision here.